workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/Commands/Create.cs

```diff
@@ -0,0 +1,191 @@
+            var devAuthTokenIdFlag = new JObject
+            {
+                { "name", $"{options.FlagPrefix}_simulated_players_dev_auth_token_id" },
+                { "value", dat.DevelopmentAuthenticationToken.Id }
+            };
+
+            var targetDeploymentFlag = new JObject
+            {
+                { "name", $"{options.FlagPrefix}_simulated_players_target_deployment" },
+                { "value", options.TargetDeployment }
+            };
+
+            var launchConfigRaw = File.ReadAllText(options.LaunchJsonPath);
+            dynamic launchConfig = JObject.Parse(launchConfigRaw);
+
+            for (var i = 0; i < launchConfig.workers.Count; ++i)
+            {
+                if (launchConfig.workers[i].worker_type == options.SimulatedCoordinatorWorkerType)
+                {
+                    launchConfig.workers[i].flags.Add(devAuthTokenIdFlag);
+                    launchConfig.workers[i].flags.Add(targetDeploymentFlag);
+                }
+            }
+
+            return launchConfig.ToString();
+        }
+
+        private static string UploadSnapshot(SnapshotServiceClient client, string snapshotPath, string projectName,
+            string deploymentName)
+        {
+            if (!File.Exists(snapshotPath))
+            {
+                Ipc.WriteError(Ipc.ErrorCode.FileNotFound, $"Could not find snapshot file at: {snapshotPath}");
+                return null;
+            }
+            // Read snapshot.
+            var bytes = File.ReadAllBytes(snapshotPath);
+
+            if (bytes.Length == 0)
+            {
+                Ipc.WriteError(Ipc.ErrorCode.Other, $"Snapshot file at {snapshotPath} has zero bytes.");
+                return null;
+            }
+
+            // Create HTTP endpoint to upload to.
+            var snapshotToUpload = new Snapshot
+            {
+                ProjectName = projectName,
+                DeploymentName = deploymentName
+            };
+
+            using (var md5 = MD5.Create())
+            {
+                snapshotToUpload.Checksum = Convert.ToBase64String(md5.ComputeHash(bytes));
+                snapshotToUpload.Size = bytes.Length;
+            }
+
+            var uploadSnapshotResponse =
+                client.UploadSnapshot(new UploadSnapshotRequest { Snapshot = snapshotToUpload });
+            snapshotToUpload = uploadSnapshotResponse.Snapshot;
+
+            // Upload content.
+            var httpRequest = WebRequest.Create(uploadSnapshotResponse.UploadUrl) as HttpWebRequest;
+            httpRequest.Method = "PUT";
+            httpRequest.ContentLength = snapshotToUpload.Size;
+            httpRequest.Headers.Set("Content-MD5", snapshotToUpload.Checksum);
+
+            using (var dataStream = httpRequest.GetRequestStream())
+            {
+                dataStream.Write(bytes, 0, bytes.Length);
+            }
+
+            // Block until we have a response.
+            httpRequest.GetResponse();
```

Comment:
Anything useful to do with the response?
Also, I believe you need to `Dispose` the returned response object

Reply:
Should check the error code for non-200 responses.